### PR TITLE
Allow recursive queries

### PR DESCRIPTION
**Pull request by lbargaoanu**
For EFCore.

### CODE REVIEW

File: src/AutoMapper/AdvancedConfiguration.cs [hunk 1]
```diff
@@ -40,5 +40,7 @@ public class AdvancedConfiguration
         public List<IExpressionResultConverter> QueryableResultConverters { get; } = ExpressionBuilder.DefaultResultConverters();
 
         public List<IExpressionBinder> QueryableBinders { get; } = ExpressionBuilder.DefaultBinders();
+
```
Comment (jbogard) [inline src/AutoMapper/AdvancedConfiguration.cs]: I think this needs XML comments

File: src/AutoMapper/MapperConfiguration.cs [hunk 2]
```diff
@@ -86,7 +87,9 @@ public void Validate(ValidationContext context)
 
         public IEnumerable<IExpressionResultConverter> ResultConverters { get; }
 
-        public IEnumerable<IExpressionBinder> Binders { get; }
+        public IEnumerable<IExpressionBinder> Binders { get; }
+        
+        public int RecursiveQueriesMaxDepth { get; }
```
Comment (lbargaoanu) [inline src/AutoMapper/MapperConfiguration.cs]: A little late perhaps, but maybe we should move the `ProjectTo` configuration to an inner object? I think this stuff doesn't even warrant a major version bump. Just sayin' :)

File: src/AutoMapper/AdvancedConfiguration.cs [hunk 3]
```diff
@@ -40,5 +40,11 @@ public class AdvancedConfiguration
...
+        /// <summary>
+        /// How many levels deep should recursive queries be expanded.
+        /// Must be zero for EF6. Can be greater than zero for EF Core.
+        /// </summary>
+        public int RecursiveQueriesMaxDepth { get; set; }
```
Comment (lbargaoanu) [inline src/AutoMapper/AdvancedConfiguration.cs]: A little late perhaps, but maybe we should move the `ProjectTo` configuration to an inner object? I think this stuff doesn't even warrant a major version bump. Just sayin' :)

Comment (jbogard) [reply src/AutoMapper/AdvancedConfiguration.cs]: Ah yeah, a little late...

**Pull request merged**